File: core/MajorityElement/MajorityElement.py
from typing import List


def majorityElement(nums: List[int]) -> int:
    num, count, length = nums[0], 1, len(nums)

    for i in range(1, length):
        if count == 0:
            num = nums[i]
            count = 1
        elif nums[i] == num:
            count += 1
        else:
            count -= 1

    return num

File: core/MajorityElement/MajorityElement.java
package com.dbc;

public class MajorityElement {
    public int majorityElement(int[] nums) {
        int num = nums[0], count = 1;
        for (int i = 1; i < nums.length; i++) {
            if (count == 0) {
                count = 1;
                num = nums[i];
            } else if (num == nums[i]) {
                count++;
            } else {
                count--;
            }
        }
        return num;
    }
}
